src/mongo/Config.class.php

```diff
@@ -2026,6 +2064,48 @@ public function getTransactionLogDatabase($readPreference = \MongoClient::RP_PRI
     }
 
     /**
+    * Creates a specification revision subdocument for a composite specification.
+    *
+    * @param \MongoCollection $specification - the composite specification
+    * @return array|null - the subdocument to add to the composite
+    **/
+    public static function getSpecRevisionSubDoc($specification) {
+        if(isset($specification) && isset($specification[_ID_KEY]) && isset($specification[_REVISION])) {
+            return array(
+                _SPEC_KEY => array(
+                    _SPEC_TYPE=> $specification[_ID_KEY],
```

Comment:
Hmm... you're right, this is loose.

I picked up the terminology from the `_id` inside the composite document.  In the tests, for example, it looks like this:

```php
            "_id"=>array(
                _ID_RESOURCE=>"http://talisaspire.com/resources/filter1",
                _ID_CONTEXT=>'http://talisaspire.com/',
                "type"=>"v_resource_filter1"),
            _SPEC_KEY => array(
                _SPEC_TYPE => "v_resource_filter1",
                _SPEC_REVISION => 1
            ),
```

So `_id.type` and `_spec.type` would be identical.  I wasn't sure whether I could just add to the `_id` field, to save duplicating it.  In either case, I'll take advice on the naming here.